vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java

```diff
@@ -475,6 +476,73 @@ void withCondition_predicateMatched_getsComponents() {
         Assertions.assertIterableEquals(List.of(div2, div3), result);
     }
 
+    @Test
+    void withText_exactMatch_getsCorrectComponent() {
+        TextComponent span1 = new TextComponent("sample text");
+        TextComponent span2 = new TextComponent("other text");
+        TextComponent span3 = new TextComponent(null);
+
+        UI.getCurrent().getElement().appendChild(span1.getElement(),
+                span2.getElement(), span3.getElement());
+
+        Assertions.assertSame(span1,
+                $(TextComponent.class).withText("sample text").findComponent());
+        Assertions.assertSame(span2,
+                $(TextComponent.class).withText("other text").findComponent());
+
+        Assertions.assertTrue($(TextComponent.class).withText("text")
+                .allComponents().isEmpty());
+        Assertions.assertTrue($(TextComponent.class).withText("SAMPLE TEXT")
+                .allComponents().isEmpty());
+        Assertions.assertTrue($(TextComponent.class).withText("other TEXT")
+                .allComponents().isEmpty());
+    }
+
+    @Test
+    void withText_null_getsAllComponent() {
+        TextComponent span1 = new TextComponent("sample text");
+        TextComponent span2 = new TextComponent("other text");
+
+        UI.getCurrent().getElement().appendChild(span1.getElement(),
+                span2.getElement());
+
+        Assertions.assertIterableEquals(List.of(span1, span2),
+                $(TextComponent.class).withText(null).allComponents());
+    }
```

Comment:
This is perhaps a non test as it tests that if we don't set anything everything works as by default.

Reply:
removed test